lib/matplotlib/artist.py

```diff
@@ -903,6 +903,27 @@ def matchfunc(x):
             artists.append(self)
         return artists
 
+    def get_zdata(self, event):
+        """
+        Get the zdata for a given event, as a string message
+        """
+        return ''
```

Comment:
I agree with @efiring on this one.  I can think of a lot of use cases (e.g. interactive sampling) where it would be very convenient to access a "raw" value.  Returning a numerical value would make this method useful for a lot more than just display.

You already have a separate method for the formatting, so it's only slightly more verbose to flip the function calls on their head and make the displayed string using `artist.format_zdata(artist.get_zdata())`.  In my opinion, this would be a better approach overall.